### PR TITLE
request/response finding pattern fix

### PR DESCRIPTION
**Pull request by future-pirate-king**

### CODE REVIEW

File: app/utils/parse-vulnerable-api-finding.ts [hunk 1]
```diff
@@ -54,27 +50,69 @@ function initializeVulnerableApiFinding(): VulnerableApiFinding {
...
       url: '',
+      cookies: {},
     },
     severity: '',
     confidence: '',
     description: '',
   };
 }
 
+function updateSection(
```
Comment (future-pirate-king) [inline app/utils/parse-vulnerable-api-finding.ts:69]: can we move the function to its original location and have an updated ts doc like earlier

File: app/components/public-api-docs/index.ts [hunk 2]
```diff
@@ -11,6 +11,7 @@ import parseError from 'irene/utils/parse-error';
 import type NetworkService from 'irene/services/network';
 import type MeService from 'irene/services/me';
 import type IntlService from 'ember-intl/services/intl';
+import { waitForPromise } from '@ember/test-waiters';
```
Comment (future-pirate-king) [inline app/components/public-api-docs/index.ts:14]: group app and lib imports together

File: app/utils/parse-vulnerable-api-finding.ts [hunk 3]
```diff
@@ -298,80 +317,136 @@ function updateFindingField(
...
-    currentHeaders = finding.response.headers;
-  } else if (key === 'params') {
-    currentSection = 'params';
-  } else if (currentSection === 'params') {
-    updateFindingParams(key, value, finding);
-  } else if (
-    currentSection === 'request-headers' ||
-    currentSection === 'response-headers'
+  if (
+    (key === 'headers' && (value === '' || value === '{}')) ||
```
Comment (Yibaebi) [inline app/utils/parse-vulnerable-api-finding.ts:425]: Can you put `value === '' || value === '{}'` in a variable?

File: app/utils/parse-vulnerable-api-finding.ts [hunk 4]
```diff
@@ -188,15 +234,15 @@ function processFirstLine(
...
+    const match = beforeColon.match(genericUrlRegex);
 
     if (match) {
-      finding.request.url = finding.request.url = firstLine
-        .substring(0, colonIndex)
-        .trim();
+      finding.url = firstLine.substring(0, colonIndex).trim();
     }
 
     if (!firstLine.startsWith('confidence:')) {
```
Comment (Yibaebi) [inline app/utils/parse-vulnerable-api-finding.ts:248]: Can we merge the two if statements? 

`if (!firstLine.startsWith('confidence:') && colonIndex !== -1)`

**Review by Yibaebi — CHANGES_REQUESTED** [review]

**Pull request merged**